Program.cs

```diff
@@ -603,6 +617,69 @@ public async void StartValidation()
     }
   }
 
+  /// <summary>
+  /// Helper class to handle the CLI options and arguments.
+  /// It is based on the CommandLine library.
+  /// </summary>
+  public class CLIParser {
+    ParserResult<CLIOptions> cliOptions;
+
+    /// <summary>
+    /// Data storage class to store the command line options and arguments.
+    /// </summary>
+    public class CLIOptions
+    {
+      [Option('s', "scope_dir", Required = false, HelpText = "Set the scope directory")]
+      public string ScopeDir {get; set;}
+
+      [Value(0, MetaName = "resource_file", HelpText = "The resource file to validate")]
+      public string ResourceFile { get; set; }
+    }
+
+    /// <summary>
+    /// Instantiate with the arguments from the command line.
+    /// <param name="args">The list of command line arguments as passed to the application</param>
+    /// </summary>
+    public CLIParser(string[] args)
+    {
+      cliOptions = Parser.Default.ParseArguments<CLIOptions>(args);
+    }
+
+    /// <summary>
+    /// Truw whether the command line was succesfully parsed, false on error.
+    /// </summary>
+    public bool success {

```

Comment:
How about `parsedSuccessfully` - and then we wouldn't need a comment, it would be self-explanatory

Reply:
Yeah, fine by me.